modules/tftp/server.rb

```diff
@@ -90,21 +107,86 @@ def pxe_default
     end
 
     def pxeconfig_file(mac)
-      ["#{pxeconfig_dir}/menu.lst.01" + mac.delete(':').upcase, "#{pxeconfig_dir}/01-" + mac.tr(':', '-').upcase]
+      ["#{pxeconfig_dir}/menu.lst.01" + mac.delete(':').upcase, "#{pxeconfig_dir}/01-" + dashed_mac(mac).upcase]
     end
   end
 
   class Pxegrub2 < Server
-    def pxeconfig_dir
-      "#{path}/grub2"
+    def bootloader_path(os, version, arch)
+      unless (bootloader_universe = Proxy::TFTP::Plugin.settings.bootloader_universe)
+        logger.debug "TFTP: bootloader universe not configured."
```

Comment:
Wouldn't this be a warning? If this code path is hit, it's probably a misconfiguration. It should be behind a capability, so Foreman shouldn't be calling it.

Reply:
This piece of code is not about Foreman calling the TFTP proxy to retrieve files. It's about the TFTP proxy deciding what files to deploy for a host to the `host-config` directory.
We check here if a user has configured the bootloader universe (which basically will only be the case if a customer wants to use Secure Boot). In case a user did not configure the bootloader universe, searching the bootloader universe can be aborted here and the default bootloader files from the `:tftproot:/grub2` directory will be deployed to the `host-config` directory.

Note: This is outdated as now the `bootloader-universe` is a dedicated directory inside the TFTP root.